rotkehlchen/externalapis/etherscan.py

```diff
@@ -236,6 +241,55 @@ def _query(
 
         return result
 
+    @staticmethod
+    def _get_or_create_genesis_transaction(
+            database: DBHandler,
+            account: ChecksumEthAddress,
+    ) -> EthereumTransaction:
+        """Add 0x0 ethereum address to DB if it is missing"""
+        db_ethx = DBEthTx(database)
+
+        # Fetch the genesis txn from DB
+        result = db_ethx.get_ethereum_transactions(
+            filter_=ETHTransactionsFilterQuery.make(
+                tx_hash=deserialize_evm_tx_hash(GENESIS_HASH),
+            ),
+            has_premium=False,
+        )
+        if len(result) == 1:
+            return result[0]
+
+        # Add the 0x0 transaction hash to known transactions
+        transaction = EthereumTransaction(
+            tx_hash=deserialize_evm_tx_hash(GENESIS_HASH),
+            block_number=0,
+            timestamp=Timestamp(1438269973),
+            from_address=deserialize_ethereum_address(GENESIS_HASH),
+            to_address=None,
+            value=0,
+            gas=0,
+            gas_price=0,
+            input_data=b'',
+            gas_used=0,
+            nonce=0,
+        )
+
+        # Add the 0x0 parent txn to DB with relevant_address as
+        # the first address to have transaction in genesis block
+        db_ethx.add_ethereum_transactions(
+            ethereum_transactions=[transaction],
+            relevant_address=account,
+        )
+        return transaction
+
+    @staticmethod
+    def _latest_trace_id(database: 'DBHandler') -> int:
+        cursor = database.conn.cursor()
+        trace_id, = cursor.execute(
+            'SELECT MAX(trace_id) from ethereum_internal_transactions',
```

Comment:
Just noticed. Don't you think that you should filter by tx hash here and get MAX only from genesis blocks trace ids?

Reply:
Take a look [here](https://github.com/rotki/rotki/blob/develop/rotkehlchen/db/schema.py#L451). Primary key is formed based on parent tx hash and trace id. Also we know that trace id is not smth shared among transactions. This is why I think that using only genesis block's transaction's trace id will make sense. Or am I missing something?

Reply:
I mean here you take MAX from all trace ids in the db but trace id is specific for each transaction